### PR TITLE
metadata: provided XMP Metadata extractors and setters

### PR DESCRIPTION
**Pull request by kucjac**
- Defined XMP Metadata examples to get and set predefined and custom models.
- Updated go.mod github.com/unidoc/unipdf/v3 to v3.31.0

### CODE REVIEW

File: metadata/pdf_get_custom_xmp_metadata.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"github.com/unidoc/unipdf/v3/core"
+	"github.com/unidoc/unipdf/v3/model"
+	"github.com/unidoc/unipdf/v3/model/xmputil"
+
+	xmprights "github.com/trimmer-io/go-xmp/models/xmp_rights"
+)
+
+const licenseKey = `
+-----BEGIN UNIDOC LICENSE KEY-----
+Free trial license keys are available at: https://unidoc.io/
```
Comment (gunnsth) [inline metadata/pdf_get_custom_xmp_metadata.go:19]: Please update to metered key, all our examples use that.  (Can copy paste from other examples)

**Review by gunnsth — COMMENTED** [review]
Looks great, except please update the license to the metered key like all the other examples use (loading it from an environment var consistently) also.  That is the most convenient for newcomers with a free key to get started.

File: metadata/pdf_get_custom_xmp_metadata.go (hunk 1 above)
Comment (kucjac) [reply metadata/pdf_get_custom_xmp_metadata.go]: Ah, I've actually copied from the only one that didn't use metered license.
Fixed all examples that had a normal license.

File: metadata/pdf_get_xmp_media_management_metadata.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+const licenseKey = `
+-----BEGIN UNIDOC LICENSE KEY-----
+Free trial license keys are available at: https://unidoc.io/
+-----END UNIDOC LICENSE KEY-----
+`
+
+const usage = "Usage: %s INPUT_PDF_PATH\n"
+
+func init() {
+	err := license.SetLicenseKey(licenseKey, `Company Name`)
```
[the remark this answers is not in the document]
Comment (kucjac) [reply metadata/pdf_get_xmp_media_management_metadata.go]: Done.

**Review by gunnsth — APPROVED** [review]
LGTM

**Pull request merged**